lib/ansible/plugins/inventory/foreman.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (C) 2016 Guido Günther <agx@sigxcpu.org>, Daniel Lobato Garcia <dlobatog@redhat.com>
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: forman
+    plugin_type: inventory
+    short_description: foreman inventory source
+    description:
+        - Get inventory hosts from the foreman service.
+    options:
+      url:
+        description: TODO
+      user:
+        description: TODO
+      password:
+        description: TODO
+      ssl_verify:
+        description: TODO
+      group_prefix:
+        description: TODO
```

Comment:
`want_facts`, `want_hostcollections`, `rich_params` are missing. Also there is no option to change the cache max age?

Reply:
will add those, cache is something I want to implement at a 'global' level for all inventory plugins, i'll add cache control options once that is done